### PR TITLE
change routing for study view page

### PR DESCRIPTION
**Pull request by dippindots**
Change routing for study view page
Fix https://github.com/cBioPortal/cbioportal/issues/5789.

### CODE REVIEW

File: src/pages/studyView/StudyViewUtils.tsx [hunk 1]
```diff
@@ -1397,4 +1398,13 @@ export function getClinicalDataCountWithColorByCategoryCounts(yesCount:number, n
         }
     }
     return getClinicalDataCountWithColorByClinicalDataCount(_.values(dataCountSet));
+}
+
+export function getStudyViewTabId(pathname:string) {
```
Comment (zhx828) [inline src/pages/studyView/StudyViewUtils.tsx]: We have the code available in results view(ResulstViewPageHelper) or patient view(no sure the file name). Could you just generalize that code and put to Util class in the shared module? @alisman do we have any Util class already in the portal related to path?

Comment (dippindots) [reply src/pages/studyView/StudyViewUtils.tsx]: Yes, they are pretty much the same. I can make a util function for them.

File: src/pages/studyView/StudyViewPage.tsx [hunk 2]
```diff
@@ -83,7 +84,7 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
                     return;
                 }
 
-                this.store.updateCurrentTab(props.routing.location.query.tab);
+                this.store.updateCurrentTab(getStudyViewTabId(getBrowserWindow().globalStores.routing.location.pathname));
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPage.tsx]: Maybe the old format should to be updated to the new structure. /study?id=***&tab=summary should be redirected to /study/summary?id=***
@alisman what's your experience in terms of handling old URL? Do we generally ignore it or redirect the tab?

**Comment by dippindots** [comment]
@alisman Rebased to the lastest master and the e2e failing seems not related to the changes in this pr.

**Comment by alisman** [comment]
@jjgao merging these with 4 failing tests unrelated

**Pull request merged**